framework/encode/deferred_operation.cpp

```diff
@@ -0,0 +1,119 @@
+/*
+** Copyright (c) 2021 Advanced Micro Devices, Inc. All rights reserved.
+**
+** Permission is hereby granted, free of charge, to any person obtaining a
+** copy of this software and associated documentation files (the "Software"),
+** to deal in the Software without restriction, including without limitation
+** the rights to use, copy, modify, merge, publish, distribute, sublicense,
+** and/or sell copies of the Software, and to permit persons to whom the
+** Software is furnished to do so, subject to the following conditions:
+**
+** The above copyright notice and this permission notice shall be included in
+** all copies or substantial portions of the Software.
+**
+** THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+** IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+** FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+** AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+** LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING
+** FROM, OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER
+** DEALINGS IN THE SOFTWARE.
+*/
+
+#include "project_version.h"
+
+#include "encode/vulkan_capture_manager.h"
+
+#include "encode/vulkan_handle_wrapper_util.h"
+#include "encode/vulkan_state_writer.h"
+#include "format/format_util.h"
+#include "generated/generated_vulkan_struct_handle_wrappers.h"
+#include "graphics/vulkan_device_util.h"
+#include "util/compressor.h"
+#include "util/file_path.h"
+#include "util/logging.h"
+#include "util/page_guard_manager.h"
+#include "util/platform.h"
+#include "encode/deferred_operation_manager.h"
+#include "encode/deferred_operation_create_ray_tracing_pipelines.h"
+
+#include <cassert>
+#include <unordered_set>
+#include <encode/struct_pointer_encoder.h>
+
+GFXRECON_BEGIN_NAMESPACE(gfxrecon)
+GFXRECON_BEGIN_NAMESPACE(encode)
+std::unique_ptr<DeferredOperationManager> DeferredOperationManager::instance_ =
```

Comment:
I recommend you make this a bare pointer to be consistent with other singleton instances like `static VulkanCaptureManager* instance_;` in `framework/encode/vulkan_capture_manager.h`.  Another possibility is to make it a `shared_ptr`, so no one has to guess about what happens after this reference is returned.

Reply:
ok, will change to shared_ptr.